packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -762,11 +811,29 @@ export class GarbageCollector implements IGarbageCollector {
         // Basically, it was written in the current GC version.
         if (result.wasSummaryTracked) {
             this.latestSummaryGCVersion = this.currentGCVersion;
+            if (this.trackGCBlobState) {
+                this.latestSerializedSummaryState = this.pendingSerializedSummaryState;
+                this.pendingSerializedSummaryState = undefined;
+            }
             return;
         }
-        // If the summary was not tracked by this client, update latest GC version from the snapshot in the result as
-        // that is now the latest summary.
-        await this.updateSummaryGCVersionFromSnapshot(result.snapshot, readAndParseBlob);
+        // If the summary was not tracked by this client, update latest GC version and blob from the snapshot in the
+        // result as that is now the latest summary.
+        const snapshot = result.snapshot;
+        const metadataBlobId = snapshot.blobs[metadataBlobName];
+        if (metadataBlobId) {
+            const metadata = await readAndParseBlob<IContainerRuntimeMetadata>(metadataBlobId);
+            this.latestSummaryGCVersion = getGCVersion(metadata);
+        }
+
+        const gcSnapshotTree = snapshot.trees[gcTreeKey];
+        if (gcSnapshotTree !== undefined && this.trackGCBlobState) {
+            const latestGCState = await getGCStateFromSnapshot(
+                gcSnapshotTree,
+                readAndParseBlob,
+            );
+            this.latestSerializedSummaryState = JSON.stringify(generateSortedGCState(latestGCState));
```

Comment:
Do we need to reset the pending state here?

Reply:
Then the `latestSerializedSummaryState` would be updated. As the summary in step 1 should be newer, then we should be using that data. If we somehow get the acks out of order that seems like a different problem.